frontend/public/components/secrets/create-secret.tsx

```diff
@@ -27,14 +29,16 @@ export enum SecretTypeAbstraction {
   webhook = 'webhook',
 }
 
-const secretDisplayType = (abstraction: SecretTypeAbstraction) => {
+const secretDisplayType = (isCreate: boolean, abstraction: SecretTypeAbstraction, t: TFunction) => {
   switch (abstraction) {
     case 'generic':
-      return 'Key/Value';
+      return isCreate ? t('public~Create Key/Value Secret') : t('public~Edit Key/Value Secret');
     case 'image':
-      return 'Image Pull';
+      return isCreate ? t('public~Create Image Pull Secret') : t('public~Edit Image Pull Secret');
     default:
-      return _.upperFirst(abstraction);
+      return isCreate
+        ? t('public~Create {{secretType}} Secret', { secretType: _.upperFirst(abstraction) })
+        : t('public~Edit {{secretType}} Secret', { secretType: _.upperFirst(abstraction) });
```

Comment:
The use of upperFirst probably needs to be updated to match the capitalization standards.